httpd/filters.go

```diff
@@ -0,0 +1,147 @@
+package httpd
```

Comment:
To be clear, I do like the movement of this code into distinct functions, which set some of the common HTTP headers. But if it's going to be a distinct file, I feel like the functions should have more a more general signature, and their implementation not rely *so* tightly on the handler implementation. Of course, others may certainly differ.